packages/beacon-node/src/sync/range/utils/peerBalancer.ts

```diff
@@ -82,14 +103,20 @@ export class ChainPeersBalancer {
     return undefined;
   }
 
-  private filterPeers(batch: Batch, requestColumns: number[], checkActiveRequest: boolean): PeerInfoColumn[] {
+  private filterPeers(batch: Batch, requestColumns: number[], noActiveRequest: boolean): PeerInfoColumn[] {
     const eligiblePeers: PeerInfoColumn[] = [];
 
     for (const peer of this.peers) {
       const {earliestAvailableSlot, custodyGroups, target, peerId} = peer;
 
       const activeRequest = this.activeRequestsByPeer.get(peerId) ?? 0;
-      if (checkActiveRequest && activeRequest > 0) {
+      if (noActiveRequest && activeRequest > 0) {
+        // consumer wants to find peer with no active request, but this peer has active request
+        continue;
+      }
+
+      if (!noActiveRequest && activeRequest >= this.maxConcurrentRequests) {
```

Comment:
```suggestion
      if (activeRequest >= this.maxConcurrentRequests) {
```
we should never return a peer gte the max